internal/driver/driver.go

```diff
@@ -428,21 +374,23 @@ func (d *Driver) handleWriteCommands(devName string, p protocolMap, reqs []dsMod
 		llrpResp = &llrp.CustomMessage{}
 
 	case ResourceReaderConfig:
-		data, err := params[0].StringValue()
+		/// Object value types come in as a map[string]interface{} which need to be
+		// marshaled back to JSON
+		reqData, err = json.Marshal(params[0].Value)
```

Comment:
Have you guys looked at using [mapstructure](https://github.com/mitchellh/mapstructure) in order to handle these conversions both here and in the app service? It seems like a better solution than going from map->json->struct.

Reply:
@ajcasagrande , no we had not, but I like it!!! THX!
@marcpfuller , the `mapstructure` package is already an approved /used package (indirectly) for EdgeX. Lets take advantage of it.

Reply:
went back to using json marshal/unmarshal